WcaOnRails/app/controllers/teams_controller.rb

```diff
@@ -1,6 +1,8 @@
 class TeamsController < ApplicationController
   before_action :authenticate_user!
-  before_action -> { redirect_unless_user(:can_edit_teams?) }
+  before_action -> { redirect_unless_user(:can_manage_teams?) }, except: [:edit, :update]
+  before_action :assign_team, only: [:edit, :update]
+  before_action -> { redirect_unless_user(:can_edit_team?, @team) }, only: [:edit, :update]
```

Comment:
I saw this `*_from_params` in other controllers and I thought what would better. Actually using it in this case would produce one unnecessary query for edit and update actions, but I will change this as you prefer the mentioned way.